crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1481,6 +1512,12 @@ impl AuthorityPerEpochStore {
                 .builder_digest_to_checkpoint
                 .insert(&digest, &sequence)?;
         }
+        let epoch_start_configuration = EpochStartConfiguration {
+            epoch_start_timestamp_ms: summary.timestamp_ms,
```

Comment:
Is there a reason that epoch start timestamp has to be the same as the first checkpoint, or is it just convenient to use it?

Reply:
This one is `put_genesis_checkpoint_in_builder` - it uses genesis checkpoint timestamp as initial timestamp of current epoch.

For other epochs `epoch_start_timestamp_ms` is actually timestamp of the last checkpoint. (But this logic is not changed in this PR, EpochStartConfiguration::epoch_start_timestamp_ms just takes existing value from the chain and caches it in the store)

Reply:
I think that instead of setting epoch start config here, a cleaner way is to pass genesis info into the constructor of epoch store, which then sets it. Doing so we also get rid of the None case. What do you think?

Reply:
I thought about it, however we can't just pass genesis data here, we need to first check if current epoch is epoch 0, and pass None otherwise. I chose to put this data in insert_genesis_checkpoint because it already has this check